File: polyglot/runtime_template_resolver/py/src/runtime_template_resolver/missing_handler.py
from typing import Optional
from .interfaces import MissingStrategy
from .errors import MissingValueError

def handle_missing(key: str, strategy: MissingStrategy = MissingStrategy.EMPTY, default_value: Optional[str] = None) -> str:
    if strategy == MissingStrategy.KEEP:
        return f"{{{{{key}}}}}"
    elif strategy == MissingStrategy.ERROR:
        raise MissingValueError(f"Missing value for placeholder: {key}")
    elif strategy == MissingStrategy.DEFAULT:
        return default_value if default_value is not None else ""
    else: # EMPTY
        return ""

File: polyglot/runtime_template_resolver/mjs/src/missing-handler.ts
import { MissingStrategy } from './interfaces.js';
import { MissingValueError } from './errors.js';

export function handleMissing(key: string, strategy: MissingStrategy = MissingStrategy.EMPTY, defaultValue?: string): string {
    switch (strategy) {
        case MissingStrategy.KEEP:
            return `{{${key}}}`;
        case MissingStrategy.ERROR:
            throw new MissingValueError(`Missing value for placeholder: ${key}`);
        case MissingStrategy.DEFAULT:
            return defaultValue ?? '';
        case MissingStrategy.EMPTY:
        default:
            return '';
    }
}
